packages/boot/test/unit/mixins/boot.mixin.unit.ts

```diff
@@ -0,0 +1,134 @@
+
+  describe('basic function presence tests', () => {
+    it('has .boot() function', () => {
+      expect(myApp.boot).to.be.a.Function();
+    });
+
+    it('has .bootControllers() function', () => {
+      expect(myApp.bootControllers).to.be.a.Function();
+    });
+
+    it('has .bootClassArtifacts() function', () => {
+      expect(myApp.bootClassArtifacts).to.be.a.Function();
+    });
+
+    it('has .bindClassArtifacts() function', () => {
+      expect(myApp.bindClassArtifacts).to.be.a.Function();
+    });
+
+    it('has .bindClassArtifactsUsingFunction() function', () => {
+      expect(myApp.bindClassArtifactsUsingFunction).to.be.a.Function();
+    });
+  });
+
+  it('binds controllers automatically', async () => {
+    const expected = [
+      'controllers.AdminController',
+      'controllers.HelloController',
+      'controllers.WorldController',
+    ];
+
+    await myApp.start();
+
+    checkBindingsArr('controllers', expected);
+    checkBindingInstance('controllers.HelloController', HelloController);
+    checkBindingInstance('controllers.WorldController', WorldController);
+    checkBindingInstance('controllers.AdminController', AdminController);
+  });
+
+  it('allows other class artifacts to be booted using prefix', () => {
+    const expected = ['repositories.TestRepository'];
+
+    myApp.bootClassArtifacts('repositories', '.repository.js', 'repositories');
+
+    checkBindingsArr('repositories', expected);
+    checkBindingInstance('repositories.TestRepository', TestRepository);
+  });
+
+  it('allows other class artifacts to be booted using function', () => {
+    const expected = ['repositoryUsingFunc.TestRepository'];
+    function bindRepoClass(inst: Constructor<{}>) {
+      myApp.bind(`repositoryUsingFunc.${inst.name}`).toClass(inst);
+    }
+
+    myApp.bootClassArtifacts('repositories', '.repository.js', bindRepoClass);
+
+    checkBindingsArr('repositoryUsingFunc', expected);
+    checkBindingInstance('repositoryUsingFunc.TestRepository', TestRepository);
+  });
+
+  it('discovers classes in given files and binds to a given prefix', () => {
+    const files = [resolve(rootDir, 'repositories/test.repository.js')];
+    const expected = ['repoPrefix.TestRepository'];
+
+    myApp.bindClassArtifacts(files, 'repoPrefix');
+
+    checkBindingsArr('repoPrefix', expected);
+    checkBindingInstance('repoPrefix.TestRepository', TestRepository);
+  });
+
+  it('discovers classes in given files and calls function to bind', () => {
+    const files = [resolve(rootDir, 'repositories/test.repository.js')];
+    const expected = ['repoFunc.TestRepository'];
+    function bindFunc(inst: Constructor<{}>) {
+      myApp.bind(`repoFunc.${inst.name}`).toClass(inst);
+    }
+
+    myApp.bindClassArtifactsUsingFunction(files, bindFunc);
+
+    checkBindingsArr('repoFunc', expected);
+    checkBindingInstance('repoFunc.TestRepository', TestRepository);
+  });
+
+  function checkBindingInstance(key: string, inst: Constructor<{}>) {
```

Comment:
repeated definition in here and the integration one. They should probably be imported from a `utils.test` file or something similar. Same thing for other helper functions